src/sourmash/index.py

```diff
@@ -124,8 +126,55 @@ def gather(self, query, *args, **kwargs):
         return results
 
     @abstractmethod
-    def select(self, ksize=None, moltype=None):
-        ""
+    def select(self, ksize=None, moltype=None, scaled=None, num=None,
+               abund=None, containment=None):
+        """Return Index containing only signatures that match requirements.
+
+        Current arguments can be any or all of:
+        * ksize
+        * moltype
+        * scaled
+        * num
+        * containment
+
+        'select' will raise ValueError if the requirements are incompatible
+        with the Index subclass.
+
+        'select' may return an empty object or None if no matches can be
+        found.
+        """
+
+
+def select_signature(ss, ksize=None, moltype=None, scaled=0, num=0,
+                     containment=False):
+    "Check that the given signature matches the specificed requirements."
+    # ksize match?
+    if ksize and ksize != ss.minhash.ksize:
+        return False
+
+    # moltype match?
+    if moltype and moltype != ss.minhash.moltype:
+        return False
+
+    # containment requires scaled; similarity does not.
+    if containment:
+        if not scaled:
+            raise ValueError("'containment' requires 'scaled' in Index.select'")
+        if not ss.minhash.scaled:
+            return False
+
+    # 'scaled' and 'num' are incompatible
+    if scaled:
```

Comment:
What's the reasoning for not checking the value of `scaled` here?

Reply:
Oooh, great question! I have two answers :). One is pragmatic, and one is conceptual.

The pragmatic answer is, we didn't need it in order for all the tests to pass 😆 

The conceptual answer is, this selects signatures that *could* be converted, but doesn't actually do the conversion. This is a tension that I haven't figured out how to resolve - should selectors return appropriately downsampled signatures, or should they just select signatures that _could_ be downsampled?

I think the right thing to do is to punt this to a new issue for discussion, since I don't think we have any situations in the codebase that depend on answering it right now (see first point, "all tests pass").

Reply:
(I will take care of punting to the new issue, but would appreciate pushback and/or your thoughts!)

Reply:
> The conceptual answer is, this selects signatures that could be converted, but doesn't actually do the conversion. This is a tension that I haven't figured out how to resolve - should selectors return appropriately downsampled signatures, or should they just select signatures that could be downsampled?

I like the approach of selecting sigs that _could_ be converted! ...but not all scaled sigs can be downsampled (desired scaled < sig scaled), right? 

>I think the right thing to do is to punt this to a new issue for discussion, since I don't think we have any situations in the codebase that depend on answering it right now (see first point, "all tests pass").
😆

Reply:
> > The conceptual answer is, this selects signatures that could be converted, but doesn't actually do the conversion. This is a tension that I haven't figured out how to resolve - should selectors return appropriately downsampled signatures, or should they just select signatures that could be downsampled?
> 
> I like the approach of selecting sigs that _could_ be converted! ...but not all scaled sigs can be downsampled (desired scaled < sig scaled), right?

yep. I think this is a great idea for a targeted test, I'll see what I can do!

Reply:
I think I prefer returning without explicit downsampling, and maybe make another function that uses `select_signatures` to do the actual downsampling.

(depending on the application that can be a lot of downsampling without really needing it...)